mypy/dmypy.py

```diff
@@ -114,16 +115,21 @@ class BadStatus(Exception):
     pass
 
 
-def main() -> None:
+def main(argv: List[str]) -> None:
     """The code is top-down."""
-    args = parser.parse_args()
+    args = parser.parse_args(argv)
     if not args.action:
         parser.print_usage()
     else:
         try:
             args.action(args)
         except BadStatus as err:
             sys.exit(err.args[0])
+        except Exception:
+            # We do this explicitly to avoid exceptions percolating up
+            # through mypy.api invocations
+            traceback.print_exc()
+            sys.exit(2)
```

Comment:
I like this choice, I think there are still some situations where 1 can be returned on a traceback in the server, but I will fix those separately later.